Assets/Scripts/PlayerMovement.cs

```diff
@@ -1,47 +1,46 @@
 ﻿using UnityEngine;
 using System.Collections;
+using System.Collections.Generic;
 using UnityEngine.SceneManagement;        //Allows us to use SceneManager
 
 //Player inherits from MovingObject, our base class for objects that can move, Enemy also inherits from this.
 public class PlayerMovement : MovingObject
 {
-    public float restartLevelDelay = 1f;        //Delay time in seconds to restart level.
-    public int pointsPerFood = 10;                //Number of points to add to player food points when picking up a food object.
-    public int pointsPerSoda = 20;                //Number of points to add to player food points when picking up a soda object.
-    // public int wallDamage = 1;                    //How much damage a player does to a wall when chopping it.
-
-
-    // private Animator animator;                    //Used to store a reference to the Player's animator component.
-    private int food;                            //Used to store player food points total during level.
-
+    private Vector3 lastPosition;
 
     //Start overrides the Start function of MovingObject
     protected override void Start()
     {
-        //Get a component reference to the Player's animator component
-        // animator = GetComponent<Animator>();
-
-        //Get the current food point total stored in GameManager.instance between levels.
-        // food = GameManager.instance.playerFoodPoints;
-
         //Call the Start function of the MovingObject base class.
         base.Start();
     }
 
-
-    //This function is called when the behaviour becomes disabled or inactive.
-    private void OnDisable()
-    {
-        //When Player object is disabled, store the current local food total in the GameManager so it can be re-loaded in next level.
-        // GameManager.instance.playerFoodPoints = food;
-    }
-
+    // protected void OnTriggerEnter2D(Collider2D other)
+    // {
+    //     if (other.gameObject.CompareTag("Enemy"))
+    //     {
+    //         Restart();
+    //     }
+    // }
 
     private void Update()
```

Comment:
Since Update is a very generic function which have many functions doing completely unrelated things in it, we should split the logic into smaller functions and only call them in the Update function instead